### PR TITLE
[cli] Provide  `export` command to export transaction to csv

### PR DESCRIPTION
**Pull request by jolestar**
```bash
sid export --start-height 864790 --end-height 864791
```

```csv
Transaction Hash,Staking Output Index,Inclusion Height,Staker Public Key,Staking Time,Finality Provider Public Key,Is Overflow,Staking Value
1b42ce46130a1d4b3bdd56b5cb325976851af1ab76951565aa4858c7d16dad00,0,864790,139f4e3ec192e83b9c6789ff644261b8fa5d7b716d1813bee744e3472f264d99,64000,fa7496f63a857d894aa393767325bf6f84560e9141f4ec54496c50f546f48bfb,false,1905000
3ddfb76b9971b786fc798a98f8fc5edc42a074c47ef28df812a389c16536b401,0,864790,b18ac73a57e6d3413284d1c91c14744464d71f19397c8ab053bc99c1ed96cafe,64000,bb0bceda25d82f10a69feca9c076d85f61d750c9a481b8105d8389325538fdd1,false,500000
967939f50dfbef960a40422843f5e3d749ceb0fba7a94542028f0dfb3e95e501,0,864790,d7c9ab7d619ada29535baec2ff77308b9014ef1c285a09fb175aa1c3e6d73f90,64000,68db13ba45241bcc1db1e63d998d78d9fce4b3f42772ef355e007a1105100476,false,500000000
27f7e4995c85ce1dc634bc6d838a7f126232a7d2459dc9f572abae56fbb0f502,0,864790,d68f068a7f1889bea28c2706bba437459eba45e9d44c49665585850be4b32e55,64000,609b4b8e27e214fd830e69a83a8270a03f7af356f64dde433a7e4b81b2399806,false,999994180
d3cb7879444d85543f73f1c51b16bbb5837383ff99d30af04d4f79a6e4a52703,0,864790,7c2e13aed3377c7781c2bd501deb638ba5ed02ad1bf757348004cf71f306f7ed,64000,db9160428e401753dc1a9952ffd4fa3386c7609cf8411d2b6d79c42323ca9923,false,26500000
```

Table

| Transaction Hash | Staking Output Index | Inclusion Height | Staker Public Key | Staking Time | Finality Provider Public Key | Is Overflow | Staking Value |
|------------------|----------------------|------------------|-------------------|--------------|------------------------------|-------------|---------------|
| 1b42ce46130a1d4b3bdd56b5cb325976851af1ab76951565aa4858c7d16dad00 | 0 | 864790 | 139f4e3ec192e83b9c6789ff644261b8fa5d7b716d1813bee744e3472f264d99 | 64000 | fa7496f63a857d894aa393767325bf6f84560e9141f4ec54496c50f546f48bfb | false | 1905000 |
| 3ddfb76b9971b786fc798a98f8fc5edc42a074c47ef28df812a389c16536b401 | 0 | 864790 | b18ac73a57e6d3413284d1c91c14744464d71f19397c8ab053bc99c1ed96cafe | 64000 | bb0bceda25d82f10a69feca9c076d85f61d750c9a481b8105d8389325538fdd1 | false | 500000 |
| 967939f50dfbef960a40422843f5e3d749ceb0fba7a94542028f0dfb3e95e501 | 0 | 864790 | d7c9ab7d619ada29535baec2ff77308b9014ef1c285a09fb175aa1c3e6d73f90 | 64000 | 68db13ba45241bcc1db1e63d998d78d9fce4b3f42772ef355e007a1105100476 | false | 500000000 |

Screenshot

Screenshot 2024-10-15 at 22 16 26

### CODE REVIEW

File: cmd/sid/cli/export.go [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+		cli.Uint64Flag{
+			Name:  "start-height",
+			Usage: "Start block height for exported transactions",
+			Value: 0,
+		},
+		cli.Uint64Flag{
+			Name:  "end-height",
+			Usage: "End block height for exported transactions",
+			Value: ^uint64(0),
+		},
```
Comment (gitferry) [inline cmd/sid/cli/export.go:41]: I was thinking these could be replaced by 3 arguments as they seem required, i.e., `sid export [start-height] [end-height] [ouput-path]`

**Review by gitferry — COMMENTED** [review]
I think this is a nice feature to have. Thanks for adding this! Could you also update the feature list in README? I would also be good if @RafilxTenfen can take a look

File: cmd/sid/cli/export.go (hunk 1 above)
Comment (jolestar) [reply cmd/sid/cli/export.go]: fixed

**Comment by jolestar** [comment]
@gitferry Update the readme.

File: cmd/sid/cli/export.go [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+		cli.Uint64Flag{
+			Name:  "start-height",
+			Usage: "Start block height for exported transactions",
+			Value: 0,
+		},
+		cli.Uint64Flag{
+			Name:  "end-height",
+			Usage: "End block height for exported transactions",
+			Value: ^uint64(0),
```
Comment (RafilxTenfen) [inline cmd/sid/cli/export.go:35]:
```suggestion
			Value: 0,
```

**Review by RafilxTenfen — COMMENTED** [review]
Can we add a test for this? But overall looks pretty good ^^

**Comment by jolestar** [comment]
@RafilxTenfen I manually test in my local env ^^.

**Review by gitferry — COMMENTED** [review]
Thanks for the update! One major comment is that the `start-height` and `end-height` should be required as args. Please check how the `btc-headers` cmd is implemented https://github.com/babylonlabs-io/staking-indexer/blob/f655d9b998c51da30f0e14b254aa7e43d9a14cd1/cmd/sid/cli/btc_headers.go#L34.

Regarding the data, could you confirm that the exported data is in the increasing order of the inclusion height?

File: cmd/sid/cli/export.go [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+	"github.com/urfave/cli"
+
+	"github.com/babylonlabs-io/staking-indexer/config"
+	"github.com/babylonlabs-io/staking-indexer/indexerstore"
+	"github.com/babylonlabs-io/staking-indexer/utils"
+)
+
+var ExportCommand = cli.Command{
+	Name:  "export",
+	Usage: "Export transactions from the indexer store to a CSV file based on block height.",
```
[the remark this answers is not in the document]
Comment (jolestar) [reply cmd/sid/cli/export.go]: fixed

File: cmd/sid/cli/export.go [hunk 4]
```diff
@@ -0,0 +1,125 @@
...
+			Usage: "Start block height for exported transactions",
+			Value: 0,
+		},
+		cli.Uint64Flag{
+			Name:  "end-height",
+			Usage: "End block height for exported transactions",
+			Value: 0,
+		},
+		cli.StringFlag{
+			Name:  "output",
```
[the remark this answers is not in the document]
Comment (jolestar) [reply cmd/sid/cli/export.go]: Reuse the flag.

File: README.md [hunk 5]
```diff
@@ -100,6 +101,22 @@ earliest `activation_height`. If the database is not empty, the user can specify
...
 
+### 5. Exporting transactions
+
+We can export the indexed transactions via command:
+
+```bash
+sid export --start-height <start-height> --end-height <end-height> --output transactions.csv
+```
+
+```txt
```
[the remark this answers is not in the document]
Comment (jolestar) [reply README.md]: I try the table and png, and finally used the png.

**Review by gitferry — APPROVED** [review]
Thanks for all the changes. LGTM!

**Comment by gitferry** [comment]
Will merge the PR after CI issue is resolved

**Comment by maiquanghiep** [comment]
Hi @jolestar, thanks for the PR.
May I ask for your favor to bump the [docker_pipeline](https://github.com/babylonlabs-io/staking-indexer/blob/main/.github/workflows/ci.yml#L21) to `v0.8.0`?
Have just updated the workflow to fix the CI issue

**Pull request merged**